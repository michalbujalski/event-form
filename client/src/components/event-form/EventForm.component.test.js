import React from "react";
import { render } from "@testing-library/react";
import EventForm from "./EventForm.component";
import { Simulate } from "react-dom/test-utils";

describe("Event form", () => {
  it("does not show errors when form not touched", () => {
    const updateTouched = jest.fn();
    const updateForm = jest.fn();
    const form = {
      firstName: "",
      lastName: "",
      email: "",
      date: new Date(),
    };
    const touched = {
      firstName: false,
      lastName: false,
      email: false,
    };
    const { queryByText } = render(
      <EventForm
        form={form}
        updateForm={updateForm}
        touched={touched}
        updateTouched={updateTouched}
      />
    );
    const emailError = queryByText(/Email must be valid/i);
    expect(emailError).toBeNull();

    const firstNameError = queryByText(/First name can't be blank/i);
    expect(firstNameError).toBeNull();

    const lastNameError = queryByText(/Last name can't be blank/i);
    expect(lastNameError).toBeNull();

    const dateError = queryByText(/Date can't be blank/i);
    expect(dateError).toBeNull();
  });

  it("renders email error", () => {
    const updateTouched = jest.fn();
    const updateForm = jest.fn();
    const form = {
      firstName: "first name",
      lastName: "last name",
      email: "",
      date: new Date(),
    };
    const touched = {
      firstName: true,
      lastName: true,
      email: true,
    };
    const { getByText } = render(
      <EventForm
        form={form}
        updateForm={updateForm}
        touched={touched}
        updateTouched={updateTouched}
      />
    );
    const element = getByText(/Email must be valid/i);
    expect(element).toBeInTheDocument();
  });
  it("renders first name error", () => {
    const updateTouched = jest.fn();
    const updateForm = jest.fn();
    const form = {
      firstName: "",
      lastName: "last name",
      email: "aaa@bbb.ccc",
      date: new Date(),
    };
    const touched = {
      firstName: true,
      lastName: true,
      email: true,
    };
    const { getByText } = render(
      <EventForm
        form={form}
        updateForm={updateForm}
        touched={touched}
        updateTouched={updateTouched}
      />
    );
    const element = getByText(/First name can't be blank/i);
    expect(element).toBeInTheDocument();
  });
  it("renders last name error", () => {
    const updateTouched = jest.fn();
    const updateForm = jest.fn();
    const form = {
      firstName: "First name",
      lastName: "",
      email: "aaa@bbb.ccc",
      date: new Date(),
    };
    const touched = {
      firstName: true,
      lastName: true,
      email: true,
    };
    const { getByText } = render(
      <EventForm
        form={form}
        updateForm={updateForm}
        touched={touched}
        updateTouched={updateTouched}
      />
    );
    const element = getByText(/Last name can't be blank/i);
    expect(element).toBeInTheDocument();
  });
  it("renders date error", () => {
    const updateForm = jest.fn();
    const updateTouched = jest.fn();
    const form = {
      firstName: "First name",
      lastName: "Last name",
      email: "aaa@bbb.ccc",
      date: "",
    };

    const touched = {
      firstName: true,
      lastName: true,
      email: true,
    };
    const { getByText } = render(
      <EventForm
        form={form}
        updateForm={updateForm}
        touched={touched}
        updateTouched={updateTouched}
      />
    );
    const element = getByText(/Date can't be blank/i);
    expect(element).toBeInTheDocument();
  });
  it("is disabled when submitting", () => {
    const updateTouched = jest.fn();
    const updateForm = jest.fn();
    const form = {
      firstName: "First name",
      lastName: "Last name",
      email: "aaa@bbb.ccc",
      date: "2020-20-03",
    };
    const touched = {
      firstName: true,
      lastName: true,
      email: true,
    };
    const { getByTestId } = render(
      <EventForm
        form={form}
        updateForm={updateForm}
        touched={touched}
        updateTouched={updateTouched}
        isLoading={true}
      />
    );
    expect(getByTestId("first-name")).toHaveAttribute("disabled");
    expect(getByTestId("last-name")).toHaveAttribute("disabled");
    expect(getByTestId("email")).toHaveAttribute("disabled");
    expect(getByTestId("date")).toHaveAttribute("disabled");
  });
  it("should submit", () => {
    const submitForom = jest.fn().mockImplementation(() => {});
    const form = {
      firstName: "First name",
      lastName: "Last name",
      email: "aaa@bbb.ccc",
      date: "2020-20-03",
    };
    const touched = {
      firstName: true,
      lastName: true,
      email: true,
    };
    const { getByText } = render(
      <EventForm form={form} updateForm={jest.fn()} submitForom={submitForom} />
    );
    Simulate.click(getByText("Create"));
    expect(submitForom).toBeCalled();
  });
  it("valid form umatches snapshot", () => {
    const submitForom = jest.fn().mockImplementation(() => {});
    const form = {
      firstName: "First name",
      lastName: "Last name",
      email: "aaa@bbb.ccc",
      date: "2020-20-03",
    };
    const touched = {
      firstName: true,
      lastName: true,
      email: true,
    };
    const component = render(
      <EventForm
        form={form}
        updateForm={jest.fn()}
        submitForom={submitForom}
        touched={touched}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
