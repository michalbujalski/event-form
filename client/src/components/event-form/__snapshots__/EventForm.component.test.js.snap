// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event form valid form umatches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root-1912"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
            >
              <div
                class="MuiBox-root MuiBox-root-2016"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    First name
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      data-testid="first-name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
            >
              <div
                class="MuiBox-root MuiBox-root-2074"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    Last name
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      data-testid="last-name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
            >
              <div
                class="MuiBox-root MuiBox-root-2075"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                  >
                    Email
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      data-testid="email"
                      type="email"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
            >
              <div
                class="MuiBox-root MuiBox-root-2076"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                    for="date-picker-inline"
                    id="date-picker-inline-label"
                  >
                    Date picker inline
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                      data-testid="date"
                      id="date-picker-inline"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        aria-label="change date"
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-2107"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Create
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-1912"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
          >
            <div
              class="MuiBox-root MuiBox-root-2016"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                >
                  First name
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    data-testid="first-name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
          >
            <div
              class="MuiBox-root MuiBox-root-2074"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                >
                  Last name
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    data-testid="last-name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
          >
            <div
              class="MuiBox-root MuiBox-root-2075"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                >
                  Email
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    data-testid="email"
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
          >
            <div
              class="MuiBox-root MuiBox-root-2076"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="date-picker-inline"
                  id="date-picker-inline-label"
                >
                  Date picker inline
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                    data-testid="date"
                    id="date-picker-inline"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      aria-label="change date"
                      class="MuiButtonBase-root MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiBox-root MuiBox-root-2107"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Create
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
